pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -233,6 +233,18 @@ def auto_reduce_results_on_epoch_end(self) -> None:
 
         self.has_reduced = True
 
+    def reset(self) -> None:
+        """
+        Call at the end of epoch to reset Result objects
+        """
+        for dl_idx in range(self.num_dataloaders):
+            epoch_metrics = self._internals[dl_idx] if not self.has_reduced else self._internals_reduced[dl_idx]
+            if self._internal_type == ResultStoreType.INSIDE_BATCH_TRAIN_LOOP:
+                for opt_idx in list(epoch_metrics):
```

Comment:
do you need the list casting here?
```suggestion
                for opt_idx in epoch_metrics:
```

Reply:
just reused the same logic from here:
https://github.com/PyTorchLightning/pytorch-lightning/blob/30b7440e123d5a91806b493995293e97a2ab80f8/pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py#L195-L199
so not sure if needed

Reply:
It is not. In the snippet you shared, `epoch_metrics` gets modified inside the loop (with `del`). That's why we make the cast to the dict keys.